compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -54,6 +54,9 @@ struct AstValidator<'a> {
 
     in_const_trait_impl: bool,
 
+    /// Are we inside a const trait defn?
+    in_const_trait_defn: bool,
```

Comment:
I don't think we need two separate boolean flags for this. We should merge it

Reply:
Got it - I combined them into `in_const_trait` - let me know if I should name it differently (pushing new commit in a moment)

Reply:
I think we should name it as `in_const_impl_or_trait`